### PR TITLE
Location category incentive validations on update template

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Add validation for updates to location, category and incentive.
## What is the intent behind these changes?

Add validation for updates to location, category and incentive.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt [hunk 1]
```diff
@@ -2,17 +2,25 @@ package uk.gov.justice.digital.hmpps.visitscheduler.utils
 
 import org.springframework.beans.factory.annotation.Value
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt:3]: Should this be in the uk.gov.justice.digital.hmpps.visitscheduler.validators package

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt [hunk 2]
```diff
@@ -121,4 +141,49 @@ class UpdateSessionTemplateValidator(
 
     return errorMessages
   }
+
+  private fun validateUpdateSessionLocation(existingSessionTemplate: SessionTemplateDto, updateSessionDetails: SessionDetailsDto, hasFutureBookedVisits: Boolean): String? {
+    // if a session has booked visits all locations should be accommodated post update
+    val errorMessage = "Cannot update locations to the new location list as all existing locations in session template are not catered for."
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt:147]: Make this a constant or put it in the after the:
return "Cannot update...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt [hunk 3]
```diff
@@ -121,4 +141,49 @@ class UpdateSessionTemplateValidator(
...
+        return errorMessage
+      }
+    }
+
+    return null
+  }
+
+  private fun validateUpdateSessionCategory(existingSessionTemplate: SessionTemplateDto, updateSessionDetails: SessionDetailsDto, hasFutureBookedVisits: Boolean): String? {
+    // if a session has booked visits all categories should be accommodated post update
+    val errorMessage = "Cannot update categories to the new category list as all existing prisoner categories in session template are not catered for."
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt:162]: same as above

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt [hunk 4]
```diff
@@ -121,4 +141,49 @@ class UpdateSessionTemplateValidator(
...
+        return errorMessage
+      }
+    }
+
+    return null
+  }
+
+  private fun validateUpdateSessionIncentiveLevels(existingSessionTemplate: SessionTemplateDto, updateSessionDetails: SessionDetailsDto, hasFutureBookedVisits: Boolean): String? {
+    // if a session has booked visits all categories should be accommodated post update
+    val errorMessage = "Cannot update incentive levels to the new incentive levels list as all existing incentive levels in session template are not catered for."
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt:177]: same as above

**Review by AledWEvans — COMMENTED** [review]
Some minor comments

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt]: Updated.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt (hunk 3 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt]: Updated.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt (hunk 4 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt]: Updated.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt:3]: Updated.

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**